src/EnergyPlus/Photovoltaics.cc

```diff
@@ -856,55 +856,53 @@ namespace Photovoltaics {
 		//       AUTHOR         B. Griffith
 		//       DATE WRITTEN   Jan. 2004
 		//       MODIFIED       B. Griffith, Aug. 2008
-		//       RE-ENGINEERED  na
 
 		// PURPOSE OF THIS SUBROUTINE:
 		// collect statements that assign to variables tied to output variables
 
-		// METHODOLOGY EMPLOYED:
-		// <description>
-
-		// REFERENCES:
-		// na
-
-		// USE STATEMENTS:
-		// na
 		// Using/Aliasing
 		using DataHeatBalance::Zone;
+		using DataGlobals::NumOfZones;
 		using DataSurfaces::Surface;
 		using DataHeatBalFanSys::QPVSysSource;
 		using TranspiredCollector::SetUTSCQdotSource;
-
-		// Locals
-		// SUBROUTINE ARGUMENT DEFINITIONS:
-
-		// SUBROUTINE PARAMETER DEFINITIONS:
-		// na
-
-		// INTERFACE BLOCK SPECIFICATIONS:
-		// na
-
-		// DERIVED TYPE DEFINITIONS:
-		// na
+		using InputProcessor::FindItemInList;
 
 		// SUBROUTINE LOCAL VARIABLE DECLARATIONS:
 		int thisZone; // working index for zones
+		int thisSurface; // index for PV surface
+		Array1D< bool > PVarrayZoneInitialized; // avoid doing the zone check too often
```

Comment:
Does this really persist through multiple calls into this function?  It looks like it's just a non-static local variable, so it will be cleaned back up each time this function is called.  And the `FirstTimeThru` variable is similar, it will just be true each time the function gets called again, right?

Here's what I'd like to propose.  Instead of adding your own new array here and first time flag, just add a single bool to the main PV array called `bool zoneInitialized;`.  Then you can let it do the allocations, you don't need an allocated flag, you don't need static variables, and you will actually have fewer code changes to accomplish this same task.  

If we kept it the way it currently is, you'd need to make the local variables static first, and then because they are static you'd need to clean them up in the `clear_state` function of this module.  (And if it doesn't have a `clear_state` you'd need to write one of your own and add it to the EnergyPlusFixture unit test class....)  In summary, avoid statics when possible! 😃 

Let me know if you have any issues with this.  The actual code fix here is great and I appreciate you getting it done so quickly.

Reply:
@Myoldmopar I don't think that should be a problem.  I will work on that as soon as I can.  I was thinking keeping it local would be better, but if what you have proposed is better, I'm all for that.  Thanks for the review and suggestion.  And glad that I could help on this!